### PR TITLE
support for password protected interface

### PR DESCRIPTION
**Pull request by scomoletti**
Initial support for password protected management interface in OpenVPN 2.2+

### CODE REVIEW

**Comment by scomoletti** [comment]
Hold on merging this for a bit..found a bug testing against clients with no password.

**Comment by scomoletti** [comment]
ok.. this should be good to go. flawed test.. typo in conf file.

File: openvpn-monitor.py [hunk 1]
```diff
@@ -110,6 +110,7 @@ def load_default_settings(self):
         self.vpns['Default VPN'] = {'name': 'default',
                                     'host': 'localhost',
                                     'port': '5555',
+                                    'password': 'changeme',
```
Comment (furlongm) [inline openvpn-monitor.py]: Can we remove this? By default most people don't have a password.

File: openvpn-monitor.py [hunk 2]
```diff
@@ -196,6 +197,7 @@ def _socket_recv(self, length):
     def _socket_connect(self, vpn):
         host = vpn['host']
         port = int(vpn['port'])
+        password = vpn['password']
```
Comment (furlongm) [inline openvpn-monitor.py]: What happens if vpn['password'] does not exist?

**Comment by PrzemekSkw** [comment]
Hi @furlongm, I use openvpn-monitor for a while but now I realize that is useless because of password lack.Anyone who see my vpn IP can enter my dashboard and see my clients real IP's and even locations. 
Regards,
Przemek

**Pull request closed without merging**